synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -402,7 +402,12 @@ async def _download_url(self, url: str, output_stream: BinaryIO) -> DownloadResu
                 url,
                 output_stream=output_stream,
                 max_size=self.max_spider_size,
-                headers={"Accept-Language": self.url_preview_accept_language},
+                headers={
+                    b"Accept-Language": self.url_preview_accept_language,
+                    b"User-Agent": [
+                        "Synapse (bot; +https://github.com/matrix-org/synapse)"
+                    ],
```

Comment:
This does something a little odd where it stomps over the header that gets configured from the homeserver config. I feel like this at the very least needs a comment.

Reply:
We should also note that we're omitting the Synapse version for security reasons (#4376), in case anyone's tempted to add it back in.